src/app/home/home.component.ts

```diff
@@ -11,24 +14,39 @@ export class HomeComponent implements OnInit {
   password: string;
   message: string;
   title = 'Login';
+  form: FormGroup;
+
   constructor(
+    private router: Router,
+    private fb: FormBuilder,
     public authService: AuthService
-  ) { }
+  ) { 
+    this.form = this.fb.group({
+      email: ['', [Validators.required, Validators.email]],
+      password: ['', [Validators.required]],
+    });
+    this.form.valueChanges.pipe(
```

Comment:
Is this mainly for debugging?

Reply:
eventually the forms will be dynamic and that'll change will fire a update error messages function.